File: Data-Structures-and-Algorithms/0027-RemoveElement/removeElement.py
class Solution(object):
    def removeElement(self, nums, val):
        count = 0
        for i in range(len(nums)):
            if nums[i] != val:
                nums[count] = nums[i]
                count += 1
            
        return count, nums

nums = [1, 2, 2, 2, 5, 7, 9, 11, 2, 19]
val = 2
print(Solution().removeElement(nums, val))

# O(N)
# O(1) for space (constant extra space)
# remove elements with value val, return number of left over elements

File: Data-Structures-and-Algorithms/0027-RemoveElement/removeElement.js
const removeElement = function(nums, val) {
    let count = 0;
    for (var i = 0; i < nums.length; i++) {
        if (nums[i] != val) {
            nums[count] = nums[i];
            count ++;
        }
    }
    return count, nums
};

nums = [1, 2, 2, 2, 5, 7, 9, 11, 2, 19]
val = 2
console.log(removeElement(nums, val))

// O(N)
// O(1) for space (constant extra space)
// remove elements with value val, return number of left over elements